File: latihan2/cpp/memory.cpp
#include <string>
#include <iostream>

using namespace std;

class memory : public hardware{//parent adalah hardware
    private://atribut
        int frequency, memorySize;
        std::string supportCuda;
    
    public:
        // constructor
        memory(){
            this->frequency = this->memorySize = 0;
            this->supportCuda = "blank";
        }
        
        // Setter dan Getter

        void setSupportCuda(string supportCuda){ //ini fungsinya cmn 1 atribut
			this->supportCuda = supportCuda;
		}
        string getSupportCuda(){
            return this->supportCuda;
        }

        void setFrequency(int frequency){ //ini fungsinya cmn 1 atribut
			this->frequency = frequency;
		}
        int getFrequency(){
            return this->frequency;
        }

        void setMemorySize(int memorySize){ //ini fungsinya cmn 1 atribut
			this->memorySize = memorySize;
		}
        int getMemorySize(){
            return this->memorySize;
        }

        // destructor
        ~memory(){}
};

File: latihan2/python/memory.py
from hardware import hardware

class memory(hardware):
    # atribut
    __frequency = 0
    __memorySize = 0
    __supportCuda = "blank"

    # constructor
    def __init__(self, supportCuda="blank", frequency=0, memorySize=0):
        self.__frequency = frequency
        self.__memorySize = memorySize
        self.__supportCuda = supportCuda
      
    # Setter dan Getter

    def setFrequency(self, frequency):
        self.__frequency = frequency
        
    def getFrequency(self):
        return self.__frequency 

    def setMemorySize(self, memorySize):
        self.__memorySize = memorySize
        
    def getMemorySize(self):
        return self.__memorySize 

    def setSupportCuda(self, supportCuda):
        self.__supportCuda = supportCuda
        
    def getSupportCuda(self):
        return self.__supportCuda 
